test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
I would suggest that we ignore IPv4 and IPv6 mix for now. I agree with @htuch that we should get good coverage the the current all IPv4 and IPv6 mix. We may want to add a few targeted mixed cases in the future for sanity checking, but I think that can be a later add on.